src/printer.js

```diff
@@ -114,13 +114,13 @@ function genericPrint(path, options, printPath, args) {
   }
 
   if (needsParens) {
-    parts.unshift("( ");
+    parts.unshift("(", options.calypsoSpacing ? " " : "");
```

Comment:
Wouldn't it be simplest if at the cop of the file you had:
`const calypsoSpace = options.calypsoSpace ? " " : ""`

Then in all cases where you have this ternary logic you could write something like:
`parths.unshift( "(", calypsoSpace );`